python/cudf/cudf/io/orc.py

```diff
@@ -106,70 +109,72 @@ def read_orc_metadata(path):
+        column_names = [
+            column_name.decode("utf-8") for column_name in column_names
+        ]
 
-    filepath_or_buffer, compression = ioutils.get_filepath_or_buffer(
-        path_or_data=filepath_or_buffer, compression=None, **kwargs
-    )
-    if compression is not None:
-        ValueError("URL content-encoding decompression is not supported")
-
-    # Read in statistics and unpack
-    (
-        column_names,
-        raw_file_statistics,
-        raw_stripes_statistics,
-    ) = liborc.read_raw_orc_statistics(filepath_or_buffer)
-
-    # Parse column names
-    column_names = [
-        column_name.decode("utf-8") for column_name in column_names
-    ]
-
-    # Parse statistics
-    cs = cs_pb2.ColumnStatistics()
-
-    file_statistics = {
-        column_names[i]: _parse_column_statistics(cs, raw_file_stats)
-        for i, raw_file_stats in enumerate(raw_file_statistics)
-        if columns is None or column_names[i] in columns
-    }
-    if any(
-        not parsed_statistics for parsed_statistics in file_statistics.values()
-    ):
-        return None
+        # Parse statistics
+        cs = cs_pb2.ColumnStatistics()
 
-    stripes_statistics = []
-    for raw_stripe_statistics in raw_stripes_statistics:
-        stripe_statistics = {
+        file_statistics = {
             column_names[i]: _parse_column_statistics(cs, raw_file_stats)
-            for i, raw_file_stats in enumerate(raw_stripe_statistics)
+            for i, raw_file_stats in enumerate(raw_file_statistics)
             if columns is None or column_names[i] in columns
         }
         if any(
             not parsed_statistics
-            for parsed_statistics in stripe_statistics.values()
+            for parsed_statistics in file_statistics.values()
         ):
-            return None
+            continue
         else:
-            stripes_statistics.append(stripe_statistics)
+            files_statistics.append(file_statistics)
 
-    return file_statistics, stripes_statistics
+        for raw_stripe_statistics in raw_stripes_statistics:
+            stripe_statistics = {
+                column_names[i]: _parse_column_statistics(cs, raw_file_stats)
+                for i, raw_file_stats in enumerate(raw_stripe_statistics)
+                if columns is None or column_names[i] in columns
+            }
+            if any(
+                not parsed_statistics
+                for parsed_statistics in stripe_statistics.values()
+            ):
+                continue
+            else:
+                stripes_statistics.append(stripe_statistics)
+
+    return files_statistics, stripes_statistics
 
 
 def _filter_stripes(
     filters, filepath_or_buffer, stripes=None, skip_rows=None, num_rows=None
 ):
+    # Multiple sources are passed as a list. If a single source is passed,
+    # wrap it in a list for unified processing downstream.
+    if not is_list_like(filepath_or_buffer):
+        filepath_or_buffer = [filepath_or_buffer]
+
```

Comment:
Is this redundant as it's already handled in `read_orc`?

Reply:
It is redundant if it is called from `read_orc` however this function can be invoked directly by other methods, specifically in dask, where they need to filter the stripes before invoking `read_orc` and that is the cause for this being here again. Does that make sense?

Reply:
Basically I think it needs to stay here for dask's sake

Reply:
Ah ok, didn't know about this. This makes sense.